### PR TITLE
Migrate to GH workflows

### PR DESCRIPTION
**Pull request by MartinBasti**

### CODE REVIEW

File: .github/workflows/gating.yaml
Comment (akhmanova) [inline .github/workflows/gating.yaml:112]: I am not sure I know what is it, could you please add a comment?

File: .github/workflows/gating.yaml
Comment (akhmanova) [inline .github/workflows/gating.yaml:43]: We set up the coverage checking with the `.coveragerc` file. Is coveralls better? 

Could you please add a comment about a coverage percentage it uses here Maybe we even will not need the `coveragerc` file anymore ..

File: .github/workflows/gating.yaml
Comment (akhmanova) [inline .github/workflows/gating.yaml:80]: Do we not need `basepython` with `black` anymore?

File: tox.ini
Comment (tkdchen) [inline tox.ini:41]: requirements-test.in includes:

```
flake8==3.9.2
flake8-docstrings==1.6.0
```

File: .github/workflows/gating.yaml
Comment (tkdchen) [inline .github/workflows/gating.yaml:61]: Either flake8 or black just needs to run with one Python version. Can this matrix be removed? Or, what requires this matrix setting?

Comment (MartinBasti) [reply .github/workflows/gating.yaml:61]: I can remove matrix, I kept it as matrix in case that we would need run flake/black on multiple versions.
With black it seems not needed at all, and with flake given that there is no py2, probably neither. I'll remove.

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:43]: coveralls provide results directly into PR and it has nice web interface for checking coverage diffs. Coveralls can measure overall decrease/increase in code coverage.

I'll check coveragerc, but in general coveragerc and coveralls are not mutually exclusive

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:112]: https://github.com/hadolint/hadolint/wiki/DL3041

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:76]: I don't follow

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:106]: I added comment with detailed explanation about what is ignored

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:43]: I'd keep  `coveragerc` as is

**Review by akhmanova — APPROVED** [review]
LGTM, thank you! Let's wait for @tkdchen approval.

**Review by tkdchen — APPROVED** [review]

**Pull request merged**
